third_party/libunwindstack/PeCoffInterface.cpp

```diff
@@ -252,6 +261,113 @@ bool PeCoffInterface<AddressType>::ParseOptionalHeader(uint64_t offset) {
+bool PeCoffInterface<AddressType>::ParseSectionHeaders(uint64_t offset) {
+  uint16_t num_sections = coff_header_.nsects;
+
+  coff_memory_.set_cur_offset(offset);
+
+  for (uint16_t idx = 0; idx < num_sections; ++idx) {
+    SectionHeader section_header;
+
+    // Section names in the header are always exactly 8 bytes. Longer names have to be looked
+    // up in the string table.
+    if (!coff_memory_.GetFully(static_cast<void*>(&section_header.name[0]), 8)) {
+      last_error_.code = ERROR_MEMORY_INVALID;
+      last_error_.address = coff_memory_.cur_offset();
+      Log::Error("Parsing section header failed: %s", GetErrorCodeString(last_error_.code));
+      return false;
+    }
+
+    if (!coff_memory_.Get32(&section_header.vmsize) ||
+        !coff_memory_.Get32(&section_header.vmaddr) || !coff_memory_.Get32(&section_header.size) ||
+        !coff_memory_.Get32(&section_header.offset) ||
+        !coff_memory_.Get32(&section_header.reloff) ||
+        !coff_memory_.Get32(&section_header.lineoff) || !coff_memory_.Get16(&section_header.nrel) ||
+        !coff_memory_.Get16(&section_header.nline) || !coff_memory_.Get32(&section_header.flags)) {
+      last_error_.code = ERROR_MEMORY_INVALID;
+      last_error_.address = coff_memory_.cur_offset();
+      Log::Error("Parsing section header failed: %s", GetErrorCodeString(last_error_.code));
+      return false;
+    }
+    parsed_section_headers_.emplace_back(section_header);
+  }
+  return true;
+}
+
+template <typename AddressType>
+bool PeCoffInterface<AddressType>::GetSectionName(const std::string& parsed_section_name_string,
+                                                  std::string* result) {
+  uint64_t offset = 0;
+  if (!android::base::ParseUint<uint64_t>(parsed_section_name_string.c_str(), &offset)) {
+    Log::Error("Failed to parse section name as integer: %s", parsed_section_name_string.c_str());
+    last_error_.code = ERROR_INVALID_COFF;
+    return false;
+  }
+
+  // The symbols come first and every one of them has a size of 18 bytes and we need to add
+  // this to the offset to get to the strings for the section names.
+  uint64_t file_offset = coff_header_.symoff + (18 * coff_header_.nsyms) + offset;
+
+  // Arbitrarily chosen to be large enough.
+  constexpr const uint64_t kMaxSectionNameLength = 1024;
+
+  if (!coff_memory_.ReadString(file_offset, result, kMaxSectionNameLength)) {
+    Log::Error("GetSectionName() failed when reading section name string");
+    last_error_.code = ERROR_MEMORY_INVALID;
+    last_error_.address = file_offset;
+    return false;
+  }
+  return true;
+}
+
+template <typename AddressType>
+bool PeCoffInterface<AddressType>::InitSections() {
+  for (const auto& section_header : parsed_section_headers_) {
+    Section section;
+    std::string header_name(section_header.name.data(), section_header.name.size());
+    std::string name_trimmed = header_name.substr(0, header_name.find('\0'));
+    if (name_trimmed.find('/') != 0) {
```

Comment:
Nit: The logic is a bit surprising to me. What about `if (name_trimmed[0] != '/') {`?

Reply:
Duh, of course :) Not sure anymore if this was historically grown and I was doing something different at some point where find was needed. Your suggestion is much better.